component/models/presexch/definition.go

```diff
@@ -693,10 +713,10 @@ func (pd *PresentationDefinition) applyRequirement(req *requirement, creds []*ve
 
 	exclude := map[string]struct{}{}
 
-	for k := range set {
-		if !req.isLenApplicable(len(set[k])) {
-			for desc, cID := range set[k] {
-				exclude[desc+cID] = struct{}{}
+	for _, descriptors := range satisfiedDescriptors {
```

Comment:
Something I noticed here: this code expects a single VC (and its BBS derived creds) to satisfy the correct number of input descriptors for a submission requirement that has nested submission requirements. As far as I know, this isn't quite right: the parent submission requirement should validate the number of satisfied nested submission requirements (not the number of input descriptors), and it shouldn't require that they be satisfied by a single credential.